modules/core/02-client/types/msgs_test.go

```diff
@@ -678,3 +684,155 @@ func TestMsgUpdateParamsGetSigners(t *testing.T) {
 		}
 	}
 }
+
+// TestMsgScheduleIBCClientUpgrade_NewMsgScheduleIBCClientUpgrade tests NewMsgScheduleIBCClientUpgrade
+func (suite *TypesTestSuite) TestMsgScheduleIBCClientUpgrade_NewMsgScheduleIBCClientUpgrade() {
+	testCases := []struct {
+		name                string
+		upgradedClientState exported.ClientState
+		expPass             bool
+	}{
+		{
+			"success",
+			ibctm.NewClientState(suite.chainA.ChainID, ibctesting.DefaultTrustLevel, ibctesting.TrustingPeriod, ibctesting.UnbondingPeriod, ibctesting.MaxClockDrift, clientHeight, commitmenttypes.GetSDKSpecs(), ibctesting.UpgradePath),
+			true,
+		},
+		{
+			"fail: failed to pack ClientState",
+			nil,
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		plan := upgradetypes.Plan{
+			Name:   "upgrade IBC clients",
+			Height: 1000,
+		}
+		_, err := types.NewMsgScheduleIBCClientUpgrade(
```

Comment:
I think we could not discard this and make some assertions on expected value in expPass conditional